ha/core/controllers/pcs/cluster_controller.py

```diff
@@ -38,7 +48,38 @@ def start(self) -> dict:
             ([dict]): Return dictionary. {"status": "", "msg":""}
                 status: Succeeded, Failed, InProgress
         """
-        raise HAUnimplemented("Cluster start operation is not implemented.")
+        _, _, _rc = self._execute.run_cmd(const.PCS_CLUSTER_STATUS, check_error=False)
+        if _rc != 0:
+            self._execute.run_cmd(const.PCS_CLUSTER_START, check_error=False)
+            time.sleep(60)
+            _, _, _rc = self._execute.run_cmd(const.PCS_CLUSTER_STATUS, check_error=False)
+            if _rc != 0:
+                return {"status": const.STATUSES.FAILED.value, "msg": "Cluster start operation failed"}
+
+        _res = self.node_list()
+        _res = json.loads(_res)
+
+        if _res.get("status") == const.STATUSES.SUCCEEDED.value:
+            _node_list = _res.get("msg")
+            Log.info(f"Node List : {_node_list}")
+            if _node_list is not None:
+                _node_group = [ _node_list[i:i + const.PCS_NODE_START_GROUP_SIZE] for i in range(0, len(_node_list), const.PCS_NODE_START_GROUP_SIZE)]
+
+                for _node_subgroup in _node_group:
+                    for _node_id in _node_subgroup:
+                        _res = self._controllers[const.NODE_CONTROLLER].start(_node_id)
+                        _res = json.loads(_res)
+                        if _res.get("status") == const.STATUSES.FAILED.value:
+                            msg = _res.get("msg")
+                            Log.error(f"Node {_node_id} : {msg}")
+                    time.sleep(30)
```

Comment:
How did we decide on 30 seconds here?

Reply:
Put this timeout in conf file. Make it less, may be 5 seconds and retry for 30 seconds if that is required. This base timeout should be used for all pcs operations. Retries in conf will be specific to operations.

Reply:
Actually, the Main intention of this delay is to wait till all the resources get started for the sub-group. So, I felt 30 secs in enough. Later, We can remove the delay and wait till all the resources become online.

Reply:
Sure. I will modify it as per Ajay's suggestion

Reply:
Sure

Reply:
As discussed with Ajay S, modified the implementation.